doc/SONiC_Port_Mirroring_HLD.md

```diff
@@ -0,0 +1,319 @@
+No changes are introduced in ASIC_DB.·
+### 3.2.5 COUNTER_DB
+No changes are introduced in COUNTER_DB.·
+
+## 3.3 Switch State Service Design
+### 3.3.1 Orchestration Agent
+
+Mirror Orchestration agent is modified to support this feature:
+   - Handle both SPAN and ERSPAN sessions separately·
+   - No changes to ERSPAN functionality.
+   - Configure mirror session based on CONFIG_DB parameters.
+   - Port mirror session is activated 
+   - Populates the mirror attribute SAI structures and pushes the entry to ASIC_DB.·
+
+## 3.4 SAI
+Mirror SAI interface APIs are already defined. More details about SAI API and attributes are described below SAI Spec @
+
+https://github.com/opencomputeproject/SAI/blob/master/inc/saimirror.h
+
+## 3.5 CLI
+### 3.5.1 Data Models
+Custom Yang model will be introduced for this feature.
+
+### 3.5.2 Configuration Commands
+
+Existing mirror session commands are enhanced to support this feature.
+
+    # Modify existing ERSPAN configuration as below.
+    config mirror_session add erspan <session-name> <src_ip> <dst_ip> <gre> <dscp>  [ttl] [queue]
+
+    #Configure Destination only span mirror session.
+    config mirror_session add span <session-name> <destination_ifName>
+
+    # Modify existing ERSPAN configuration to accept source port and direction
+    config mirror_session add erspan <session-name> <src_ip> <dst_ip> <gre> <dscp>  [ttl] [queue] [src_port] [rx/tx/both]
+
+    #Configure Port mirror span mirror session.
+    config mirror_session add span <session-name> <destination_ifName> <source_ifName> <rx/tx/both>
+
+
+KLISH CLI Support.
+
+    # SPAN config
+    # **switch(config)# [no] mirror-session <session-name>** <br>
+    **switch(config-mirror-<session-name>)# [no] destination <dest_ifName> [source <src_ifName> direction <rx/tx/both>]** <br>
+    dest_ifName can be port only
+    src_ifName can be port/port-channel>
+
+    # ERSPAN config
+    **switch(config)# [no] mirror-session <session-name>** <br>
+    **switch(config-mirror-<session-name>)# [no] destination erspan src_ip <src_ip> dst_ip <dst_ip> dscp < dscp > ttl < ttl > [ gre < gre >] [queue <queue>] [source <src_ifName> direction <rx/tx>**] <br>
+
+### 3.5.3 Show Commands
+
+The following show command display all the mirror sessions that are configured.
+
+    # show mirror-session
+    ERSPAN Sessions
+    ---------------------------------------------------------------------------------------------------------
+      Name     Status      SRC IP    DST IP    GRE   DSCP    TTL  Queue Policer      SRC Port    Direction
+    everflow0  active    10.1.0.32  10.0.0.7    10    10      10
+    everflow1  active    10.1.0.33  10.0.0.8    10    10      10                     Ethernet4    both
+
+    SPAN Sessions
+    ---------------------------------------------------------------------------------------------------------
+      Name Status         DST Port         SRC Port Direction
+     sess1 active        Ethernet4        Ethernet0     rx
```

Comment:
what is the definition of active status? what is the criteria. can you make it clear in the document?

Reply:
Port mirror session will be active in below cases.
1. When destination port only session is created, then once the session is created from SAI. the session becomes active. These sessions can be used in ACL mirroring.
2. When mirroring is enabled on the source ports, then the session will become active.